### PR TITLE
Add note on private registry & repo tips

### PR DESCRIPTION
**Pull request by ianshmean**
After working through these issues setting up a private registry with private repos, I thought the tips might be helpful for the readme.

This works well for me now.

Thanks for making this package!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -88,3 +88,35 @@ register(package)
...
+```
+export SSH_PUB_KEY_PATH=~/.ssh/key.pub
+export SSH_KEY_PATH=~/.ssh/key
+```
+
+### 2. Change Pkg to ssh mode
+Run the following to switch Pkg mode to the `git@github.co:...` ssh protocol
+```
+julia> Pkg.setprotocol!(protocol="ssh")
+```
```
Comment (GunnarFarneback) [inline README.md]: I've never had to do this. If you register your package with an ssh URL and have the Pkg protocol set to the default `nothing` option, I believe things should work out of the box. But if you previously have set the protocol to `https` you may need to change.

File: README.md [hunk 2]
```diff
@@ -88,3 +88,35 @@ register(package)
...
+
+## Working with a private registry and/or private repositories
+
+If the registry needs to be private and/or privately hosted packages are 
+added to it, two additional steps are required to make the user experience simple.
+
+### 1. Set up persistent git auth that Julia recognizes
+
+Julia's internal git doesn't recognize `~/.ssh/config` settings, so one way is
+to set `SSH_PUB_KEY_PATH` and `SSH_KEY_PATH` environmental variables:
```
[the remark this answers is not in the document]
Comment (ianshmean) [reply README.md]: That's good to know. I typically use differently named ssh keys, so I guess I'm an outlier

File: README.md (hunk 1 above)
Comment (ianshmean) [reply README.md]: Also good to know. I'll simplify these tips

**Comment by ianshmean** [comment]
@GunnarFarneback Thanks for the pointers. I've made the tips simpler. 

Just a suggestion btw. I just struggled so thought I'd try to help people in my shoes

**Comment by GunnarFarneback** [comment]
It's somewhat tangential to this package but I agree that it's also something people may very well run into when using it. Thanks.

**Pull request merged**